x/signutil/sign.go

```diff
@@ -0,0 +1,124 @@
+package signutil
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+)
+
+type MsgArbitrarySignature struct {
```

Comment:
Let's rename `MsgArbitrarySignature` to `MsgSignedText` so that it's clear that we're not handling binary data.

Also, Antoine claims that one of our partners has difficulty parsing JSON, so we need another Msg type that consists of an array of key/value pairs, each of type string.

```go
type Pair struct {
   Key string
   Value string
}

type MsgSignedPairs struct {
   Pairs []Pair
   Signer sdk.AccAddress
}
```
Please add a `--pairs` flag that reads a JSON array of objects with properties `key` and `value`.  If the pairs' JSON is not perfectly formed as [ { key:k0, value:v0 }, { key:k1, value:v1}, ... ] then the app should panic.